mkosi/__init__.py

```diff
@@ -6273,6 +6273,37 @@ def run_ssh(args: CommandLineArguments) -> None:
         )
 
 
+def generate_secure_boot_key(args: CommandLineArguments) -> NoReturn:
+    """Generate secure boot keys using openssl"""
+    for f in (args.secure_boot_key, args.secure_boot_certificate):
+        if f is not None and not os.path.exists(f):
+            die(
+                f"{f} already exists.\n"
+                "If you are sure you want to generate new secure boot keys\n"
+                "remove {args.secure_boot_key} and {args.secure_boot_certificate} first."
+            )
+
+    os.execvp(
+        "openssl",
+        [
+            "openssl",
+            "req",
+            "-new",
+            "-x509",
+            "-newkey",
+            "rsa:2048",
+            "-keyout",
+            args.secure_boot_key or "./mkosi.secure-boot.key",
+            "-out",
+            args.secure_boot_certificate or "./mkosi.secure-boot.crt",
+            "-days",
+            "3650",
```

Comment:
i'd probably do 2y. i.e. 1y sounds like a good time to roll them over, and let's add a bit of space around that, and doubling sounds like a safe choice for that.

Reply:
Changed the time to two years and added a note to the user, when the key expires.